File: node/src/transfers.js
'use strict';
// Registro de transferencias a terceros (LLMs, pasarelas, APIs externas).
// Enviar el mensaje de un cliente a Anthropic/OpenAI ES una comunicación a un tercero.
class TransferRegistry {
  constructor(store, config) { this.store = store; this.config = config; }

  log(subjectId, destino, finalidad, categorias = []) {
    const tp = this.config.thirdParty(destino);
    this.store.put('privacy_transfers', {
      ts: new Date().toISOString(),
      subjectId,
      destino,
      pais: tp?.pais || '?',
      baseLegal: tp?.base || '?',
      finalidad,
      categorias,
      terceroDeclarado: !!tp, // false => transferencia NO declarada en config
    });
  }
}
module.exports = { TransferRegistry };

File: privacy_kit/transfers.py
"""Registro de transferencias a terceros (LLMs, pasarelas, APIs externas).

La 21.719 exige saber a quién se ceden/comunican datos y con qué base. Mandar el
mensaje de un cliente a Anthropic/OpenAI ES una comunicación a un tercero.
"""
from __future__ import annotations
from datetime import datetime, timezone

from .config import PrivacyConfig
from .store.base import PrivacyStore


class TransferRegistry:
    def __init__(self, store: PrivacyStore, config: PrivacyConfig):
        self.store = store
        self.config = config

    def log(self, subject_id: str, destino: str, finalidad: str,
            categorias: list[str] | None = None) -> None:
        tp = self.config.third_party(destino)
        self.store.put("privacy_transfers", {
            "ts": datetime.now(timezone.utc).isoformat(),
            "subject_id": subject_id,
            "destino": destino,
            "pais": (tp or {}).get("pais", "?"),
            "base_legal": (tp or {}).get("base", "?"),
            "finalidad": finalidad,
            "categorias": categorias or [],
            "tercero_declarado": tp is not None,  # False => transferencia NO registrada en config
        })
